methods.rb

```diff
@@ -0,0 +1,116 @@
+module Enumerable
+  def my_each
+    i = 0
+    length.times do
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_map(proc = nil)
+    temp = []
+    i = 0
+    if proc.nil?
+      length.times do
+        temp.push(yield(self[i]))
+        i += 1
+      end
+    else
+      length.times do
+        temp.push(proc.call(self[i]))
+        i += 1
+      end
+    end
+    temp
+  end
+
+  def my_each_with_index
+    i = 0
+    length.times do
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    temp = []
+    i = 0
+    length.times do
+      temp.push(self[i]) if yield(self[i])
+      i += 1
+    end
+    temp
+  end
+
+  def my_all?
+    i = 0
+    length.times do
+      return false unless yield(self[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?
+    i = 0
+    length.times do
+      return true if yield(self[i])
+
+      i += 1
+    end
+    false
+  end
+
+  def my_none?
+    i = 0
+    length.times do
+      return false if yield(self[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_count(arg = nil)
+    unless arg.nil?
+      i = 0
+      count = 0
+      length.times do
+        count += 1 if self[i] == arg
+        i += 1
+      end
+      return count
+    end
+    length
+  end
+
+  def my_inject(arg = nil)
+    accumulator = arg
+    if arg.nil?
+      accumulator = self[0]
+      i = 1
+      (length - 1).times do
+        accumulator = yield(accumulator, self[i])
+        i += 1
+      end
+    else
+      i = 0
+      length.times do
+        accumulator = yield(accumulator, self[i])
+        i += 1
+      end
+    end
+    accumulator
+  end
+end
```

Comment:
- [ ] `my_inject` is not working as the original `inject` method, also is not passing all test cases.

### `inject`` is an Enumerable that has these characteristics:
- Returns a `LocalJumpError` when no block is given and no symbol is passed as an argument.
```
[1, 2, 3].inject
# LocalJumpError (no block given)
```

- it does not modify/change the original array/range/hash.

- it can accept an initial value and/or a symbol as an argument.
```
[1, 2, 3].inject(2, :*)
# 12
```

[Enumerable#inject](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-inject)